### PR TITLE
Fix asserts in Vector<>

### LINKED ISSUE #78717: [NativeAOT] System.Numerics.Vectors test fails to compile with an assert.
To repro - just build tests with chk runtime config (like `build.cmd clr+libs+libs.tests -rc Checked -lc Release /p:TestNativeAot=true`)

A more direct repro is to just compile one test: `"C:\naot02\runtime\artifacts\bin\coreclr\windows.x64.Checked\ilc-published\\ilc" @"C:\naot02\runtime\artifacts\obj\System.Numerics.Vectors.Tests\Release\net7.0\native\System.Numerics.Vectors.Tests.ilc.rsp`

The command fails with:

```
ILC: C:\naot02\runtime\src\coreclr\jit\hwintrinsic.cpp:564
ILC: Assertion failed 'genActualType(arg->gtType) == genActualType(argType)' in 'System.Numerics.Vector:Load[ulong](ulong):System.Numerics.Vector`1[ulong]' during 'Importation' (IL size 7; hash 0xfe953b2c; FullOpts)
```

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/dotnet/runtime/issues/78717 (and unblocks jit-diff)

This fixes two asserts:

1) This used to assert while it is expected to early out in importer as an unsupported type.
```csharp
Int128 Foo(Vector<Int128> v) => v[0];
```

2) Prejitting of 
```csharp
        [Intrinsic]
        [CLSCompliant(false)]
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public static Vector<T> Load<T>(T* source)
            where T : unmanaged => LoadUnsafe(ref *source);
```

Fails in that assert `assert(genActualType(arg->gtType) == genActualType(argType));`
My understanding it's completely fine that we pass ULONG (representing pointer) to BYREF

PTAL @tannergooding

### CODE REVIEW

File: src/coreclr/jit/hwintrinsic.cpp [hunk 1]
```diff
@@ -560,8 +560,6 @@ GenTree* Compiler::getArgForHWIntrinsic(var_types            argType,
 
         arg = impPopStack().val;
         assert(varTypeIsArithmetic(arg->TypeGet()) || ((argType == TYP_BYREF) && arg->TypeIs(TYP_BYREF)));
-
-        assert(genActualType(arg->gtType) == genActualType(argType));
```
Comment (jakobbotsch) [inline src/coreclr/jit/hwintrinsic.cpp:564]: I think you can replace this by `impCheckImplicitArgumentCoercion`

Comment (jakobbotsch) [reply src/coreclr/jit/hwintrinsic.cpp:564]: In fact, we should probably `BADCODE` if `impCheckImplicitArgumentCoercion` returns false, in the same way as `impPopCallArgs`

Comment (EgorBo) [reply src/coreclr/jit/hwintrinsic.cpp:564]: Oh, TIL: impCheckImplicitArgumentCoercion exists

Comment (EgorBo) [reply src/coreclr/jit/hwintrinsic.cpp:564]: Thanks, addressed

**Comment by EgorBo** [comment]
@jakobbotsch does it look good otherwise?

File: src/coreclr/jit/hwintrinsic.cpp [hunk 2]
```diff
@@ -561,7 +561,10 @@ GenTree* Compiler::getArgForHWIntrinsic(var_types            argType,
         arg = impPopStack().val;
         assert(varTypeIsArithmetic(arg->TypeGet()) || ((argType == TYP_BYREF) && arg->TypeIs(TYP_BYREF)));
 
-        assert(genActualType(arg->gtType) == genActualType(argType));
+        if (!impCheckImplicitArgumentCoercion(argType, arg->gtType))
+        {
+            BADCODE("the hwintrinsic argument has a type that can't be implicitly converted to the signature type");
+        }
```
Comment (jakobbotsch) [inline src/coreclr/jit/hwintrinsic.cpp:567]: Ideally we should have this check outside the `else` but let's not rerun CI for it

**Review by jakobbotsch — APPROVED** [review]

File: src/coreclr/jit/hwintrinsic.cpp (hunk 1 above)
Comment (srwrr) [reply src/coreclr/jit/hwintrinsic.cpp:564]: 78765

**Pull request merged**